Modules/CaPTkCommon/include/CaPTkFeatureScalingClass.cpp

```diff
@@ -0,0 +1,176 @@
+  for (unsigned int sampleNo = 0; sampleNo < NumberOfSamples; sampleNo++)
+    scaledFeatureSet(sampleNo, NumberOfFeatures) = inputdata(sampleNo, NumberOfFeatures);
+  
+  return scaledFeatureSet;
+}
+double captk::FeatureScalingClass::GetZScore(const double &mean, const double &variance, const double &featureval)
+{
+  //return (featureval - mean) / (variance + FLT_MIN);
+  return (featureval - mean) / (variance + 0.0);
+}
+void captk::FeatureScalingClass::ScaleGivenTrainingFeatures(const captk::FeatureScalingClass::VariableSizeMatrixType &inputdata, captk::FeatureScalingClass::VariableSizeMatrixType &scaledFeatureSet, VariableLengthVectorType &meanVector, VariableLengthVectorType &stdVector)
+{
+  int NumberOfSamples = inputdata.Rows();
+  int NumberOfFeatures = inputdata.Cols();
+
+  //---------calculate mean and variance for each feature----------------
+  meanVector.SetSize(NumberOfFeatures);
+  stdVector.SetSize(NumberOfFeatures);
+  for (int featureNo = 0; featureNo < NumberOfFeatures; featureNo++)
+  {
+    double temp = 0.0;
+    for (int sampleNo = 0; sampleNo < NumberOfSamples; sampleNo++)
+      temp = temp + inputdata(sampleNo, featureNo);
+    meanVector[featureNo] = temp / NumberOfSamples;
+    double mean = meanVector[featureNo];
+    temp = 0.0;
+    for (int sampleNo = 0; sampleNo < NumberOfSamples; sampleNo++)
+      temp = temp + (inputdata(sampleNo, featureNo) - mean)*(inputdata(sampleNo, featureNo) - mean);
+    double std = std::sqrt(temp / (NumberOfSamples - 1));
+    stdVector[featureNo] = std;
+  }
+  //---------calculate z-score for each feature value-----------------------------
+  scaledFeatureSet.SetSize(NumberOfSamples, NumberOfFeatures);
+  for (int featureNo = 0; featureNo < NumberOfFeatures; featureNo++)
+  {
+    for (int sampleNo = 0; sampleNo < NumberOfSamples; sampleNo++)
+      scaledFeatureSet(sampleNo, featureNo) = GetZScore(meanVector[featureNo], stdVector[featureNo], inputdata(sampleNo, featureNo));
+  }
+}
+captk::FeatureScalingClass::VariableSizeMatrixType captk::FeatureScalingClass::ScaleGivenTestingFeatures(const captk::FeatureScalingClass::VariableSizeMatrixType &inputdata, const VariableLengthVectorType &meandata, const VariableLengthVectorType &stddata)
+{
+  int NumberOfSamples = inputdata.Rows();
+  int NumberOfFeatures = inputdata.Cols();
+
+  //---------calculate z-score for each feature value-----------------------------
+  captk::FeatureScalingClass::VariableSizeMatrixType scaledFeatureSet;
+  scaledFeatureSet.SetSize(NumberOfSamples, NumberOfFeatures); //+1 to score the actual label
+
+  for (int featureNo = 0; featureNo < NumberOfFeatures; featureNo++)
+  {
+    for (int sampleNo = 0; sampleNo < NumberOfSamples; sampleNo++)
+      scaledFeatureSet(sampleNo, featureNo) = GetZScore(meandata[featureNo], stddata[featureNo], inputdata(sampleNo, featureNo));
+  }
+
+  //typedef itk::CSVNumericObjectFileWriter<double, 1, 453> WriterTypeMatrixMean;
```

Comment:
Please delete all commented code.

Reply:
The commented-out code has been removed.